### PR TITLE
Support SkipLocalsInit on event accessors

### PR DESCRIPTION
**Pull request by agocke**
Relates to https://github.com/dotnet/roslyn/issues/25780 (test plan for skip locals init)

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.cs [hunk 1]
```diff
@@ -10451,6 +10483,72 @@ void M()
...
+        }
+        remove
+        {
+            int x = 1;
+            x += x + 1;
+        }
+    }
+
+}";
+            var comp = CreateCompilation(source, options: TestOptions.DebugDll.WithAllowUnsafe(true));
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.cs]: >WithAllowUnsafe [](start = 79, length = 15)

do we need unsafe?

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.cs [hunk 2]
```diff
@@ -10451,6 +10483,72 @@ void M()
...
+  IL_0002:  stloc.0
+  IL_0003:  ldloc.0
+  IL_0004:  ldloc.0
+  IL_0005:  ldc.i4.1
+  IL_0006:  add
+  IL_0007:  add
+  IL_0008:  stloc.0
+  IL_0009:  ret
+}";
+            verifier.VerifyIL("C.E.add", il);
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.cs]: >verifier.VerifyIL("C.E.add", il); [](start = 12, length = 33)

do you want to also verify C.E2.add and remove?

File: src/Compilers/Core/Portable/Symbols/Attributes/CommonEventWellKnownAttributeData.cs [hunk 3]
```diff
@@ -1,16 +1,12 @@
...
-using System.Diagnostics;
-using Microsoft.CodeAnalysis.Text;
-
 namespace Microsoft.CodeAnalysis
 {
     /// <summary>
     /// Information decoded from well-known custom attributes applied on an event.
     /// </summary>
-    internal class CommonEventWellKnownAttributeData : WellKnownAttributeData
+    internal class CommonEventWellKnownAttributeData : WellKnownAttributeData, ISkipLocalsInitAttributeTarget
```
Comment (jcouv) [inline src/Compilers/Core/Portable/Symbols/Attributes/CommonEventWellKnownAttributeData.cs]: >ISkipLocalsInitAttributeTarget [](start = 79, length = 30)

where does the interface come into play? is it necessary?

File: src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.cs (hunk 1 above)
Comment (agocke) [reply src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.cs]: Yup, that was the compromise I decided on with Jan. It's unsafe, but only a little, so let's just require `unsafe` passed to the compiler.

File: src/Compilers/Core/Portable/Symbols/Attributes/CommonEventWellKnownAttributeData.cs (hunk 3 above)
Comment (agocke) [reply src/Compilers/Core/Portable/Symbols/Attributes/CommonEventWellKnownAttributeData.cs]: See `AttributeData.DecodeSkipLocalsInitAttribute<T>`. The interface is how we share functionality between the different AttributeData classes (like CommonEventWellKnownAttributeData, MethodWellKnownAttributeData, et al)

**Comment by agocke** [comment]
ping @dotnet/roslyn-compiler for one more review

**Comment by RikkiGibson** [comment]
Looks like there are conflicts with the base branch

**Pull request merged**